src/audio/SDL_audio.c

```diff
@@ -1109,6 +1114,52 @@ SDL_GetAudioDeviceName(int index, int iscapture)
 }
 
 
+void
+SDL_GetAudioDeviceSpec(int index, int iscapture, SDL_AudioSpec *spec)
+{
+    SDL_bool found = SDL_FALSE;
+
+    if (spec == NULL) {
+        SDL_SetError("spec cannot be NULL");
+        return;
+    }
+
+    if (!SDL_WasInit(SDL_INIT_AUDIO)) {
+        SDL_SetError("Audio subsystem is not initialized");
+        SDL_zerop(spec);
```

Comment:
Just zerop(spec) after we know it's not NULL, so we don't have to do this on each error condition (and risk forgetting one later).